cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java

```diff
@@ -614,6 +614,10 @@ public void stop(Id.Namespace id, Id.Program programId, ProgramType type) throws
       if (cause instanceof IllegalArgumentException) {
         throw (IllegalArgumentException) cause;
       }
+
```

Comment:
seems better to check if the `cause` is of type `RuntimeException`. If that's the case no need to wrap. You can simply do:

```throw Throwables.propagate(e.getCause())```